src/CachePlugin.php

```diff
@@ -20,29 +22,26 @@ class CachePlugin implements Plugin
     private $pool;
 
     /**
-     * Default time to store object in cache. This value is used if CachePlugin::respectCacheHeaders is false or
-     * if cache headers are missing.
-     *
-     * @var int
+     * @var StreamFactory
      */
-    private $defaultTtl;
+    private $streamFactory;
 
-    /**
-     * Look at the cache headers to know how long this response is going to be cached.
-     *
-     * @var bool
-     */
-    private $respectCacheHeaders;
+    private $config = [
+        'default_ttl' => null,
+        'respect_cache_headers' => true,
+    ];
 
     /**
      * @param CacheItemPoolInterface $pool
-     * @param array                  $options
+     * @param StreamFactory          $streamFactory
+     * @param array                  $config
      */
-    public function __construct(CacheItemPoolInterface $pool, array $options = [])
+    public function __construct(CacheItemPoolInterface $pool, StreamFactory $streamFactory, array $config = [])
     {
         $this->pool = $pool;
-        $this->defaultTtl = isset($options['default_ttl']) ? $options['default_ttl'] : null;
-        $this->respectCacheHeaders = isset($options['respect_cache_headers']) ? $options['respect_cache_headers'] : true;
+        $this->streamFactory = $streamFactory;
+
+        $this->config = $this->configure($config);
```

Comment:
The reason why a setter would be better (no matter if it is per options or just one accepting an array) is that we don't break BC if we need a typehinted parameter. Config/options is optional anyway.